hail_search/queries/mito.py

```diff
@@ -118,6 +122,177 @@ class MitoHailTableQuery(BaseHailTableQuery):
+                annotation=sample_type.passes_inheritance_field, entries_ht_field=sample_type.family_entries_field
+            )
+
+        family_idx_map = self._build_family_index_map(sample_types, sorted_wes_family_sample_data, sorted_wgs_family_sample_data)
+        ht = self._apply_multi_sample_type_entry_filters(ht, family_idx_map)
+        ch_ht = self._apply_multi_sample_type_entry_filters(ch_ht, family_idx_map)
+
+        return ht, ch_ht
+
+    @staticmethod
+    def _build_family_index_map(sample_types, sorted_wes_family_sample_data, sorted_wgs_family_sample_data):
+        family_guid_idx_map = defaultdict(dict)
+        for sample_type, sorted_family_sample_data in sample_types:
+            for family_idx, samples in enumerate(sorted_family_sample_data):
+                family_guid = samples[0]['familyGuid']
+                family_guid_idx_map[family_guid][sample_type.value] = family_idx
+        return hl.dict(family_guid_idx_map)
+
+    def _apply_multi_sample_type_entry_filters(self, ht, family_idx_map):
+        if ht is None:
+            return ht
+
+        # Keep family from both sample types if either passes quality AND inheritance
+        for sample_type in SampleType:
+            ht = ht.annotate(**{
+                sample_type.family_entries_field: hl.enumerate(ht[sample_type.family_entries_field]).starmap(
+                    lambda i, samples: hl.or_missing(
+                        self._family_passes_quality_inheritance(
+                            ht, sample_type, i, samples[0]['familyGuid'], family_idx_map
+                        ), samples)
+                )})
+
+        # Merge family entries and filters from both sample types
+        ht = ht.transmute(
+            family_entries=hl.coalesce(
+                ht[SampleType.WES.family_entries_field], hl.empty_array(ht[SampleType.WES.family_entries_field].dtype.element_type)
+            ).extend(hl.coalesce(
+                ht[SampleType.WGS.family_entries_field], hl.empty_array(ht[SampleType.WGS.family_entries_field].dtype.element_type)
+            )).filter(lambda entries: entries.any(hl.is_defined))
+        )
+        # Filter out families with no valid entries in either sample type
+        return ht.filter(ht.family_entries.any(hl.is_defined))
+
+    @staticmethod
+    def _family_passes_quality_inheritance(ht, sample_type, sample_type_family_idx, family_guid, family_idx_map):
+        # Note: This logic does not sufficiently handle case 2 here https://docs.google.com/presentation/d/1hqDV8ulhviUcR5C4PtNUqkCLXKDsc6pccgFVlFmWUAU/edit?usp=sharing
+        # and will need to be changed to support it.
+        return (
+            hl.is_defined(ht[sample_type.passes_quality_field][sample_type_family_idx]) &
+            hl.is_defined(ht[sample_type.passes_inheritance_field][sample_type_family_idx])
+        ) | (
+            (hl.is_defined(family_guid) & family_idx_map.get(family_guid).contains(sample_type.other_sample_type.value)) &
```

Comment:
`family_guid` is not defined if there are no calls for the family for a variant. 
![image](https://github.com/user-attachments/assets/e6581e3d-bde8-4557-938e-d0060e4366fa)

Reply:
That makes sense, but then it seems like if there was no call in the first sample but calls in the second sample this behavior would be wrong. I think rather than checking `family_guid` here it would be better to only call this helper above if theres a valid sample, and for getting the family_guid use `hl.coalesce(samples)[0]['familyGuid']`:
```
hl.or_missing(samples.any(hl.is_defined) &  self._family_passes_quality_inheritance(...
```